[GTK][WPE] Adopt std::span in webkit_web_context_set_preferred_languages()

csaavedra opened the pull request:
#### 3c4b21bf290f0c34f91c78994cb54acdbf088b9c
<pre>
[GTK][WPE] Adopt std::span in webkit_web_context_set_preferred_languages()
<a href="https://bugs.webkit.org/show_bug.cgi?id=283424">https://bugs.webkit.org/show_bug.cgi?id=283424</a>

Reviewed by Adrian Perez de Castro.

Add a helper to fill a std::span with a strv (char**) and use it
to avoid accessing a raw array when setting preferred languages
in a web context.

* Source/WTF/wtf/glib/GSpanExtras.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:

Canonical link: <a href="https://commits.webkit.org/286900@main">https://commits.webkit.org/286900@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c165e4f458cbd61aa6a9b94ab1f540e1f87dd724

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77452 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56487 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/30368 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/82033 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28734 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65635 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4784 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60693 "Found 85 new test failures: fast/block/positioning/fixed-container-with-relative-parent.html fast/css-custom-paint/out-of-memory-while-adding-worklet-module.html fast/css/focus-ring-exists-for-search-field.html fast/images/object-data-url-case-insensitivity.html fast/inline/list-marker-inside-container-with-margin.html fast/inline/min-content-width-with-hypens.html fast/inline/overflowing-content-with-hypens.html fast/mediastream/canvas-video-to-canvas.html fast/mediastream/captureStream/canvas3d.html fast/multicol/table-vertical-align.html ... (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18694 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80519 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50661 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66471 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40974 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/48063 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23967 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/27057 "Built successfully") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/70640 "Found 2 new JSC stress test failures: stress/get-by-val-hoist-above-structure-2.js.ftl-eager-no-cjit-b3o1, stress/get-by-val-hoist-above-structure.js.ftl-eager-no-cjit-b3o1 (failure)") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/69172 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/24305 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/83441 "Built successfully") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/76731 "Found 1 new JSC stress test failure: stress/attribute-custom-accessor.js.ftl-eager (failure)") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/4832 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/3281 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68936 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4988 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66439 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68200 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12200 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/10292 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/98983 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11991 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/4779 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/7594 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21636 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/4798 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/8233 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6557 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WTF/wtf/glib/GSpanExtras.h:
```diff
@@ -102,6 +102,12 @@ inline std::span<const uint8_t> span(const GRefPtr<GVariant>& variant)
     return span(variant.get());
 }
 
+static inline std::span<char*> span(char **strv)
```
Align the pointer asterisks to the left:

```suggestion
static inline std::span<char*> span(char** strv)
```

csaavedra replied:
Upsi-daisy!

csaavedra committed:
[GTK][WPE] Adopt std::span in webkit_web_context_set_preferred_languages()
https://bugs.webkit.org/show_bug.cgi?id=283424

Reviewed by Adrian Perez de Castro.

Add a helper to fill a std::span with a strv (char**) and use it
to avoid accessing a raw array when setting preferred languages
in a web context.

* Source/WTF/wtf/glib/GSpanExtras.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp:

Canonical link: https://commits.webkit.org/286900@main
